300-LongestIncreasingSubsequence/step1.cpp

```diff
@@ -0,0 +1,28 @@
+#include <vector>
+using namespace std;
+class Solution {
+ public:
+  int lengthOfLIS(vector<int>& nums) {
+    const int kLengthOfNums = nums.size();
+    vector<int> length_to_mini(kLengthOfNums + 1, 1e5);
+    length_to_mini[0] = -1e5;
+    for (int num : nums) {
+      vector<int> prev = length_to_mini;
+      for (int i = 1; i < kLengthOfNums + 1; i++) {
+        if (prev[i - 1] < num) {
+          if (num < prev[i]) {
+            length_to_mini[i] = num;
+          }
+        }
+      }
+    }
+    int _min = 1;
+    for (int i = 0; i < kLengthOfNums + 1; i++) {
+      if (length_to_mini[i] == 1e5) {
+        break;
+      }
+      _min = i;
```

Comment:
変数名の先頭を `_` とするのはあまり見かけないように思います。 `min_length` はいかがでしょうか？

補足すると、 `__` が含まれる変数、 `_` のあとに大文字が来る変数は C++ の予約語となっています。必ず避けましょう。 `_` のとに小文字が来る変数も将来的に予約語となる可能性がありますので、原則避けたほうがよいと思います。

https://learn.microsoft.com/en-us/cpp/cpp/identifiers-cpp?view=msvc-170
> Use of two sequential underscore characters ( __ ) in an identifier, or a single leading underscore followed by a capital letter, is reserved for C++ implementations in all scopes. You should avoid using one leading underscore followed by a lowercase letter for names with file scope because of possible conflicts with current or future reserved identifiers.